src/wazuh_db/wdb.c

```diff
@@ -1311,36 +1110,25 @@ int wdb_update_last_vacuum_data(wdb_t* wdb, const char *last_vacuum_time, const
 }
 
 void wdb_close_old() {
-    wdb_t * node;
-    wdb_t * next;
+    char ** keys = wdb_pool_keys();
 
-    rwlock_lock_read(&pool_mutex);
-    wdb_t *copy = wdb_pool_copy();
-    rwlock_unlock(&pool_mutex);
+    for (int i = 0; keys[i] && wdb_open_count > wconfig.open_db_limit; i++) {
+        wdb_t * node = wdb_pool_get(keys[i]);
 
-    for (wdb_t *i = copy; i != NULL; wdb_destroy(i), i = next) {
-        next = i->next;
-
-        rwlock_lock_write(&pool_mutex);
-        node = (wdb_t *)OSHash_Get(open_dbs, i->id);
-
-        if (node == NULL || db_pool_size <= wconfig.open_db_limit) {
-            rwlock_unlock(&pool_mutex);
+        if (node == NULL) {
             continue;
         }
 
-        w_mutex_lock(&node->mutex);
-
-        if (node->refcount == 0 && !node->transaction) {
-            w_mutex_unlock(&node->mutex);
```

Comment:
Just a simple doubt, it's ok that bedfore we checked for a refcount == 0 and now ==1 ?